src/storage/UpdateVertexProcessor.cpp

```diff
@@ -341,10 +341,22 @@ void UpdateVertexProcessor::process(const cpp2::UpdateVertexRequest& req) {
             return std::string("");
         },
         [this, partId, vId, req] (kvstore::ResultCode code) {
-            this->pushResultCode(this->to(code), partId);
+            cpp2::ResultCode thriftResult;
+            thriftResult.set_code(to(code));
+            thriftResult.set_part_id(partId);
+            if (code == kvstore::ResultCode::ERR_LEADER_CHANGED) {
+                nebula::cpp2::HostAddr leader;
+                auto addrRet = kvstore_->partLeader(this->spaceId_, partId);
+                CHECK(ok(addrRet));
+                auto addr = value(std::move(addrRet));
+                leader.set_ip(addr.first);
+                leader.set_port(addr.second);
+                thriftResult.set_leader(std::move(leader));
+            }
             if (code == kvstore::ResultCode::SUCCEEDED) {
                 onProcessFinished(req.get_return_columns().size());
             } else {
+                this->codes_.emplace_back(std::move(thriftResult));
```

Comment:
Why not use `pushResultCode`.

Reply:
And better use `if ... else if ... else` statements.

Reply:
Ok